wallet/chainntfns.go

```diff
@@ -889,6 +890,23 @@ func (w *Wallet) VoteOnOwnedTickets(ctx context.Context, winningTicketHashes []*
 				ticketVoteBits = tvb
 			}
 
+			// When not on mainnet, randomly disapprove blocks based
+			// on the disapprove percent.
+			dp := w.DisapprovePercent()
+			if dp > 0 {
+				if w.chainParams.Net == wire.MainNet {
+					panic("block disapproval percent set on mainnet")
+				}
+				rand.Seed(time.Now().UnixNano())
+				if int(dp) > rand.Intn(100) {
+					log.Infof("Disapproving block %v voted with ticket %v",
+						blockHash, ticketHash)
+					// Set the BlockValid bit to zero,
+					// disapproving the block.
+					ticketVoteBits.Bits &= 0
```

Comment:
Minor nit: this is setting _all_ voting bits to zero, not just the block validity bit.

Reply:
ah

Reply:
I think it's just getting the last bit now.